script/tool/test/drive_examples_command_test.dart

```diff
@@ -35,52 +39,55 @@ void main() {
       runner.addCommand(command);
     });
 
-    test('driving under folder "test"', () async {
-      final Directory pluginDirectory = createFakePlugin(
-        'plugin',
-        packagesDir,
-        extraFiles: <String>[
-          'example/test_driver/plugin_test.dart',
-          'example/test/plugin.dart',
-        ],
-        platformSupport: <String, PlatformSupport>{
-          kPlatformAndroid: PlatformSupport.inline,
-          kPlatformIos: PlatformSupport.inline,
-        },
+    void setMockFlutterDevicesOutput({
+      bool hasIosDevice = true,
+      bool hasAndroidDevice = true,
+    }) {
+      final List<String> devices = <String>[
+        if (hasIosDevice) '{"id": "$_fakeIosDevice", "targetPlatform": "ios"}',
+        if (hasAndroidDevice)
+          '{"id": "$_fakeAndroidDevice", "targetPlatform": "android-x86"}',
+      ];
+      final String output = '''[${devices.join(',')}]''';
+
+      final MockProcess mockDevicesProcess = MockProcess();
+      mockDevicesProcess.exitCodeCompleter.complete(0);
+      mockDevicesProcess.stdoutController.close(); // ignore: unawaited_futures
+      processRunner.processToReturn = mockDevicesProcess;
+      processRunner.resultStdout = output;
+    }
+
+    test('fails if no platforms are provided', () async {
+      setMockFlutterDevicesOutput();
+      expect(
+        () => runCapturingPrint(runner, <String>['drive-examples']),
+        throwsA(isA<ToolExit>()),
       );
+    });
 
-      final Directory pluginExampleDirectory =
-          pluginDirectory.childDirectory('example');
-
-      final List<String> output = await runCapturingPrint(runner, <String>[
-        'drive-examples',
-      ]);
+    test('fails for iOS if no iOS devices are present', () async {
+      setMockFlutterDevicesOutput(hasIosDevice: false);
+      expect(
+        () => runCapturingPrint(runner, <String>['drive-examples']),
+        throwsA(isA<ToolExit>()),
+      );
+    });
 
+    test('fails if no platforms are provided', () async {
```

Comment:
Shouldn't this test have a device, but fail since there is no platform provided?

Reply:
This test was copypasta'd (it was supposed to be the Android version of the test above but I never actually changed it); fixed. I also did a sweep and fixed all tests that asserted `ToolExit`s to actually check the output as well so that they aren't failing for the wrong reasons (which was true of several more tests).